src/pdfFiller/1040FieldMappings.ts

```diff
@@ -0,0 +1,115 @@
+import { Dependent, FilingStatus, IncomeW2, Information, PersonRole } from '../redux/data'
+
+interface FieldMapping {
+  [field: number]: (info: Information) => string | boolean | undefined
+}
+
+const depFields = (s: Information, depIdx: number): [string, string, string, boolean, boolean] | undefined => {
+  const deps: Dependent[] = s.taxPayer?.dependents ?? []
+
+  if (deps.length > depIdx) {
+    const dep: Dependent = deps[depIdx]
+    return [
+      dep.lastName, dep.ssid, dep.relationship, false, false
+    ]
+  }
+
+  return undefined
+}
+
+const depField = (s: Information, fieldIdx: number): string | boolean | undefined => {
+  const fs = depFields(s, Math.floor((fieldIdx - 32) / 5))
+  if (fs !== undefined) {
+    return fs[(fieldIdx - 32) % 5]
+  }
+  return undefined
+}
+
+const depFieldMappings = (): FieldMapping => {
+  return Object.fromEntries(
+    Array.from(Array(20)).map((n: number) => [n + 32, (s: Information) => depField(s, n + 32)])
+  )
+}
+
+const wages = (s: Information): number => {
+  if ((s.w2s ?? []).length < 1) {
+    return 0
+  }
+  return s.w2s.map((w2) => w2.income).reduce((l, r) => l + r)
+}
+
+const fedWithholding = (s: Information): number => {
+  if ((s.w2s ?? []).length < 1) {
+    return 0
+  }
+  return s.w2s.map((w2) => w2.fedWithholding).reduce((l, r) => l + r)
+}
+
+const w2ForRole = (s: Information, r: PersonRole): IncomeW2 | undefined =>
+  (s.w2s ?? []).find((w2) => w2.personRole === r)
+
+const fieldMappings: FieldMapping = {
+  0: (s) => s.taxPayer?.filingStatus === FilingStatus.S,
+  1: (s) => s.taxPayer?.filingStatus === FilingStatus.MFJ,
+  2: (s) => s.taxPayer?.filingStatus === FilingStatus.MFS,
+  3: (s) => s.taxPayer?.filingStatus === FilingStatus.HOH,
+  4: (s) => s.taxPayer?.filingStatus === FilingStatus.W,
+  5: (s) => '',
+  6: (s) => s.taxPayer?.primaryPerson?.firstName,
+  7: (s) => s.taxPayer?.primaryPerson?.lastName,
+  8: (s) => s.taxPayer?.primaryPerson?.ssid,
+  9: (s) => s.taxPayer?.spouse?.firstName,
+  10: (s) => s.taxPayer?.spouse?.lastName,
+  11: (s) => s.taxPayer?.spouse?.ssid,
+  12: (s) => s.taxPayer?.primaryPerson?.address?.address,
+  13: (s) => s.taxPayer?.primaryPerson?.address?.aptNo,
+  14: (s) => s.taxPayer?.primaryPerson?.address?.city,
+  15: (s) => s.taxPayer?.primaryPerson?.address?.state,
+  16: (s) => s.taxPayer?.primaryPerson?.address?.zip,
+  17: (s) => s.taxPayer?.primaryPerson?.address?.foreignCountry,
+  18: (s) => s.taxPayer?.primaryPerson?.address?.province,
+  19: (s) => s.taxPayer?.primaryPerson?.address?.postalCode,
+  20: (s) => false,
```

Comment:
These and each of the commented lines below should become issues.

Reply:
Thanks for commenting a description of each field index, that's really helpful!